Dockerfile

```
@@ -1,24 +1,41 @@
-FROM ubuntu:16.04
+FROM ubuntu:latest
 
 ARG BUILDER_UID=9999
+ARG DEBIAN_FRONTEND=noninteractive
 
 ENV PATH /home/builder/.local/bin:$PATH
+ENV PYTHON_VERSION 3.8.12
 
 RUN apt-get update && apt-get install -y --no-install-recommends \
     ca-certificates \
     git \
     python3-dev \
     wget \
+    # Pyenv pre-requisites
+    make build-essential libssl-dev zlib1g-dev \
+    libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev \
     && rm -rf /var/lib/apt/lists/*
 
-RUN wget -q https://bootstrap.pypa.io/pip/3.5/get-pip.py \
-    && python3 get-pip.py pip==18.1 \
-    && rm -rf get-pip.py
+# Set-up necessary Env vars for PyEnv
+ENV PYENV_ROOT $HOME/.pyenv
+ENV PATH $PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
 
-RUN python3 -m pip install \
+# Install pyenv
+RUN set -ex \
+    && curl https://pyenv.run | bash \
+    && pyenv install $PYTHON_VERSION \
+    && pyenv global $PYTHON_VERSION \
+    && pyenv rehash
```

need to make pyenv shims writable by all here otherwise you'll get an error ```pyenv: cannot rehash: /.pyenv/shims isn't writable``` during a build

```suggestion
    && pyenv rehash \
    && chmod -R a+w $PYENV_ROOT/shims
```